src/app.js

```diff
@@ -3,11 +3,184 @@ import $ from 'jquery';
 import _ from 'underscore';
 
 // CSS
-import './css/foundation.css';
-import './css/style.css';
+//import './css/foundation.css';
+//import './css/style.css';
+
+// Models and Collections
+import Trip from './app/models/trip';
+import TripList from './app/collections/trip_list';
+import Reservation from './app/models/reservation';
+
+// Vars that need to be defined up front
+let tripRowTemplate;
+let tripDetailsTemplate;
+let trip = new Trip();
+const tripList = new TripList();
+const tripFields = ['name', 'category', 'continent', 'weeks', 'cost', 'about'];
+const reservationFields = ['name', 'age', 'email'];
+
+// Callback functions
+const render = function render(tripList) {
+  $('#trip-list').empty();
+  tripList.forEach((trip) => {
+    $('#trip-list').append(tripRowTemplate(trip.attributes));
+  });
+};
+
+const getTrip = function getTrip() {
+  const id = $(this).attr('id');
+  trip = tripList.get(id);
+  trip.fetch({success: events.successfulGetTrip, error: events.failedGetTrip});
+};
+
+const getTripList = function getTripList() {
+  tripList.fetch({success: events.successfulGetTripList, error: events.failedGetTripList});
+};
+
+const showModal = function showModal() {
+  $('#add-form-modal').show();
+};
+
+const hideModal = function hideModal() {
+  $('#add-form-modal').hide();
+  $('#add-trip-form').trigger('reset');
+  $('#add-form-status-message').hide();
+}
+
+const events = {
+  successfulGetTrip(trip) {
+    $('#trip-not-found').hide();
+    $('#trip-info').empty();
+    $('#trip-info').append(tripDetailsTemplate(trip.attributes));
+  },
+  failedGetTrip() {
+    $('#trip-not-found').show();
+  },
+  successfulGetTripList() {
+    $('#list-not-found').hide();
+    $('#trip-table').show();
+  },
+  failedGetTripList() {
+    $('#list-not-found').show();
+  },
+  sortTrips(event) {
+    const sortingBy = $(this).attr('class').split(' ');
+    $('.sort').removeClass('current-sort-field');
+    $(this).addClass('current-sort-field');
+    if (sortingBy[1] === tripList.comparator) {
+      tripList.models.reverse();
```

Comment:
This is a temporary change -- Backbone won't "know" that we've reversed the sort order for this comparator.

As a result, when Backbone decides to re-sort the collection for us (such as when a new trip is added), it will go back to the non-reversed order. How could we implement the sorting for our Backbone collection such that it will keep track of whether the sort was reversed or not?